doc/manual/cookbooks/choosing_nonlinear_solver_scheme/doc.tex

```diff
@@ -0,0 +1,326 @@
+When the problem to solve in \aspect{} is nonlinear, choosing the right nonlinear 
+solver scheme can be crucial to get fast convergence to the correct solution or  
+even to converge at all. This section is aimed to explain the differences between 
+the three approaches available in \aspect{}: iterated Stokes, iterated defect correction
+Stokes and Newton Stokes. The actual nonlinear solver you choose also will state 
+which advection solver you choose, but we won't discuss that in this section.
```

Comment:
I think it would be good to discuss this later on. Or at least mention that we can't use iterative advection schemes when the models involving updating compositional fields with reaction terms.